demisto_sdk/commands/download/downloader.py

```diff
@@ -95,647 +87,851 @@
-        :return: The list of all custom content objects
+        Converts a mapping of file names to raw file data (StringIO),
+        into a mapping of file names to custom content objects (parsed & loaded data)
+
+        Note:
+            Custom content items with an empty 'type' key are not supported and will be omitted.
+
+        Args:
+            custom_content_data (dict[str, StringIO]): A dictionary mapping file names to their content.
+
+        Returns:
+            dict[str, dict]: A dictionary mapping content item's file names, to dictionaries containing metadata
+                about the content item, and file data.
         """
-        custom_content_file_paths: list = get_child_files(self.custom_content_temp_dir)
-        custom_content_objects: List = list()
-        for file_path in custom_content_file_paths:
+        logger.info("Parsing downloaded custom content data...")
+        custom_content_objects: dict[str, dict] = {}
+
+        for file_name, file_data in custom_content_data.items():
             try:
-                custom_content_object: Dict = self.build_custom_content_object(
-                    file_path
+                logger.debug(f"Parsing '{file_name}'...")
+                custom_content_object: Dict = self.create_content_item_object(
+                    file_name=file_name, file_data=file_data
                 )
-                if custom_content_object["type"]:
-                    # If custom content object's type is empty it means the file isn't of support content entity
-                    custom_content_objects.append(custom_content_object)
-            # Do not add file to custom_content_objects if it has an invalid format
-            except ValueError as e:
-                logger.info(f"[red]Error when loading {file_path}, skipping[/red]")
-                logger.info(f"[red]{e}[/red]")
+
+                # Check if all required fields are present
+                missing_field = False
+                for _field in ("id", "name", "entity", "type"):
+                    if not custom_content_object.get(_field):
+                        logger.warning(
+                            f"'{file_name}' will be skipped as its {_field} could not be detected."
+                        )
+                        missing_field = True
+                        break
+
+                # If the content is missing a required field, skip it
+                if missing_field:
+                    continue
+
+                # If the content is written in JavaScript (not supported), skip it
+                if custom_content_object["type"] in (
+                    FileType.INTEGRATION,
+                    FileType.SCRIPT,
+                ) and custom_content_object.get("code_lang") in (None, "javascript"):
+                    logger.warning(
+                        f"Skipping '{file_name}' as JavaScript content is not supported."
+                    )
+                    continue
```

Comment:
why do we skip javascript content?

Reply:
No idea. It's like that from the existing code, I just kept it as is as I was unsure what's the reason behind it.

https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/download/downloader.py#L981-L999